src/IROperator.cpp

```diff
@@ -400,7 +400,7 @@ Expr lossless_cast(Type t, Expr e) {
     }
 
     if (const Cast *c = e.as<Cast>()) {
-        if (t.can_represent(c->value.type())) {
+        if (c->type.can_represent(c->value.type())) {
```

Comment:
I don't think I understand this change.

Reply:
The comment says it wants to allow recursion through widening casts (= casts that don't lose information). The old code didn't actually implement that, the new code does. This is also in a more targeted change #5459 that will hopefully be merged first.